### ISSUE TITLE
[metadata.tvmaze (1.0.1)]: *** Unhandled exception detected: <class 'ValueError'> invalid literal for int() with base 10

### ISSUE THREAD

**Issue by nSpecter** [romanvm/kodi.tvmaze#14]
For a while now I've been getting an ever increasing number of errors when "Scanning TV Shows Using TVMaze".

Most are repeating, one per episode or series, I'm not entirely sure which, that look like:

ERROR: [metadata.tvmaze (1.0.1)]: *** Unhandled exception detected: <class 'ValueError'> invalid literal for int() with base 10: 'http://www.thetvdb.com/api/1D62F2F90030C444/series/78804/all/en.zip' *** 
[Redacted for brevity. See attached log file.]
WARNING: No information found for item '/home/nspectre/Flix/TV/Doctor Who (2005)/', it won't be added to the library.

The remaining sequence of errors (minus debug info, please see attached log) are:

2020-01-16 17:30:00.494 T:32596   ERROR: [metadata.tvmaze (1.0.1)]: *** Unhandled exception detected: <class 'ValueError'> invalid literal for int() with base 10: 'http://www.thetvdb.com/api/1D62F2F90030C444/series/286058/all/en.zip' ***
2020-01-16 17:30:00.994 T:32545 WARNING: No information found for item '/home/nspectre/Flix/TV/Grantchester/', it won't be added to the library.

2020-01-16 17:30:09.376 T:32609   ERROR: [metadata.tvmaze (1.0.1)]: *** Unhandled exception detected: <class 'ValueError'> invalid literal for int() with base 10: 'http://www.thetvdb.com/api/1D62F2F90030C444/series/260449/all/en.zip' ***
2020-01-16 17:30:09.998 T:32545 WARNING: No information found for item '/home/nspectre/Flix/TV/Vikings/', it won't be added to the library.

2020-01-16 17:30:20.662 T:32624   ERROR: [metadata.tvmaze (1.0.1)]: *** Unhandled exception detected: <class 'ValueError'> invalid literal for int() with base 10: 'http://www.thetvdb.com/api/1D62F2F90030C444/series/80660/all/en.zip' ***
2020-01-16 17:30:22.296 T:32545 WARNING: No information found for item '/home/nspectre/Flix/TV/The Graham Norton Show/', it won't be added to the library.

2020-01-16 17:30:36.670 T:32652   ERROR: [metadata.tvmaze (1.0.1)]: *** Unhandled exception detected: <class 'ValueError'> invalid literal for int() with base 10: 'http://www.thetvdb.com/api/1D62F2F90030C444/series/76355/all/en.zip' ***
2020-01-16 17:30:37.315 T:32545 WARNING: No information found for item '/home/nspectre/Flix/TV/Silent Witness/', it won't be added to the library.

2020-01-16 17:30:45.074 T:32667   ERROR: [metadata.tvmaze (1.0.1)]: *** Unhandled exception detected: <class 'ValueError'> invalid literal for int() with base 10: 'http://www.thetvdb.com/api/1D62F2F90030C444/series/348204/all/en.zip' ***
2020-01-16 17:30:45.757 T:32545 WARNING: No information found for item '/home/nspectre/Flix/TV/The War Of The Worlds/', it won't be added to the library.
2020-01-16 17:30:50.435 T:32545 WARNING: No information found for item '/home/nspectre/Flix/TV/American Experience/', it won't be added to the library.
2020-01-16 17:30:55.522 T:32545 WARNING: No information found for item '/home/nspectre/Flix/TV/Chicago Fire/', it won't be added to the library.

2020-01-16 17:31:05.741 T:32705   ERROR: [metadata.tvmaze (1.0.1)]: *** Unhandled exception detected: <class 'ValueError'> invalid literal for int() with base 10: 'http://www.thetvdb.com/api/1D62F2F90030C444/series/81157/all/en.zip' ***
2020-01-16 17:31:06.388 T:32545 WARNING: No information found for item '/home/nspectre/Flix/TV/Nature/', it won't be added to the library.

2020-01-16 17:31:09.149 T:32711   ERROR: [metadata.tvmaze (1.0.1)]: TVmaze returned an error: 404 Client Error: Not Found for url: http://api.tvmaze.com/lookup/shows?thetvdb=82615
2020-01-16 17:31:09.209 T:32711   ERROR: [metadata.tvmaze (1.0.1)]: *** Unhandled exception detected: <class 'TypeError'> 'NoneType' object is not subscriptable ***

2020-01-16 17:31:09.732 T:32545   ERROR: GetDirectory - Error getting plugin://metadata.tvmaze/?action=getepisodelist&url=http%3a%2f%2fwww.thetvdb.com%2fapi%2f1D62F2F90030C444%2fseries%2f82615%2fall%2fen.zip&pathSettings=%7b%7d
2020-01-16 17:31:09.733 T:32545 WARNING: No information found for item '/home/nspectre/Flix/TV/Particle Physics for Non-Physicists/', it won't be added to the library.

2020-01-16 17:31:20.316 T:32727   ERROR: [metadata.tvmaze (1.0.1)]: *** Unhandled exception detected: <class 'ValueError'> invalid literal for int() with base 10: 'http://www.thetvdb.com/api/1D62F2F90030C444/series/328634/all/en.zip' ***
2020-01-16 17:31:20.999 T:32545 WARNING: No information found for item '/home/nspectre/Flix/TV/The Good Doctor/', it won't be added to the library.

2020-01-16 17:31:30.427 T:32742   ERROR: [metadata.tvmaze (1.0.1)]: *** Unhandled exception detected: <class 'ValueError'> invalid literal for int() with base 10: 'http://www.thetvdb.com/api/1D62F2F90030C444/series/362696/all/en.zip' ***
2020-01-16 17:31:31.132 T:32545 WARNING: No information found for item '/home/nspectre/Flix/TV/The Witcher/', it won't be added to the library.

2020-01-16 17:31:40.304 T:32757   ERROR: [metadata.tvmaze (1.0.1)]: *** Unhandled exception detected: <class 'ValueError'> invalid literal for int() with base 10: 'http://www.thetvdb.com/api/1D62F2F90030C444/series/159651/all/en.zip' ***
2020-01-16 17:31:40.966 T:32545 WARNING: No information found for item '/home/nspectre/Flix/TV/How the Universe Works/', it won't be added to the library.
2020-01-16 17:31:46.090 T:32545 WARNING: No information found for item '/home/nspectre/Flix/TV/The Resident/', it won't be added to the library.
2020-01-16 17:31:51.578 T:32545 WARNING: No information found for item '/home/nspectre/Flix/TV/FBI/', it won't be added to the library.
2020-01-16 17:32:01.267 T:32545 WARNING: No information found for item '/home/nspectre/Flix/TV/Chicago P.D/', it won't be added to the library.

2020-01-16 17:32:12.663 T:333   ERROR: [metadata.tvmaze (1.0.1)]: *** Unhandled exception detected: <class 'ValueError'> invalid literal for int() with base 10: 'http://www.thetvdb.com/api/1D62F2F90030C444/series/359251/all/en.zip' ***
2020-01-16 17:32:13.395 T:32545 WARNING: No information found for item '/home/nspectre/Flix/TV/Seven Worlds, One Planet/', it won't be added to the library.
2020-01-16 17:32:13.489 T:32545  NOTICE: VideoInfoScanner: Finished scan. Scanning for video info took 02:32

I attempted to curl -G 'http://www.thetvdb.com/api/1D62F2F90030C444/series/286058/all/en.zip' and received back the response:

\<html>
\<head>\<title>301 Moved Permanently\</title>\</head>
\<body bgcolor="white">
\<center>\<h1>301 Moved Permanently\</h1>\</center>
\<hr>\<center>CloudFront\</center>
\</body>
\</html>

However, if I paste that URL into my browser it properly returns the requested zip file containing XML files for actors, banners and en.

I suspect this has to do with the recent TheTVDB discombobulation fiasco. ;)

[TVMAZE Error Log](https://github.com/romanvm/kodi.tvmaze/files/4074911/TVMAZE.Error.Log)

If I may be of any help, don't hesitate to ask. And thank you for everything. :)

**Comment by KarellenX** [comment]
It looks like you are using the TVDB `<epidoseguide>` while using TVMaze scraper. 

I guess you originally scraped your show with TVDB, now you have decided to change scrapers mid-show and are trying to add new episodes to the existing show in your library.

That won't work. 

Once you scrape a show with a specific scraper, you need to continue to use it for subsequent scrapes of new episodes of the show. If you want to change scrapers, then you need to Refresh the TV Show (no need to refesh the episodes)

Of course I could be wrong, but I wouldn't know because you did not provide a full log!!

**Comment by romanvm** [comment]
It's a known bug of Kodi (not this scraper): https://github.com/xbmc/xbmc/issues/16979

For now the solution is ether to edit your .NFO files and remove `episodeguide` section or to remove .NFO files completely.

**Comment by nSpecter** [comment]
Excellent! I see you're on it.

Thanks to both of you for the valuable input.

I did some quantification/experimentation with the new information.

I took the first broken Series(1) in the error log and removed \<episodeguideurl>blahblahblah\</episodeguideurl> from it's tvshow.nfo file.

I took the second broken Series(2) and removed \<episodeguide>blahblahblah\</episodeguide>

I took the third broken Series(3) and removed both lines. 

I then loaded Kodi and it performed its scan with the usual bunch of error notifications.

The first broken Series(1) DID come in successfully and is now in the Kodi library. Series(2) and Series(3) did NOT get scanned into the library. Examining the Kodi log file shows the Series(1)'s error has been replaced with a new error (below) and the other two do not appear to have changed. All subsequent errors have also not changed (as I have not yet edited their tvshow.nfo files)

The new error for Series(1) is:

[New Series(1) Error.txt](https://github.com/romanvm/kodi.tvmaze/files/4080158/New.Series.1.Error.txt)

Continuing testing, I changed Series(2)'s NFO file to remove the remaining \<episodeguideurl>blahblahblah\</episodeguideurl>, _effectively making it the same as Series(3) with both lines removed_. I then loaded Kodi and let it scan.

As I anticipated, Series(2) did NOT get scanned into the library. As far as I can tell it is still producing the same error message in the Kodi log file. Ditto with Series(3).

NOTE: Series(1) did NOT reproduce the new SSL error or any other error. It appears to have been successfully scanned into the library and that series is fixed.

Continuing testing, I added _back_ into the Series(2) NFO file the original \<episodeguide>blahblahblah\</episodeguide>, _effectively making it the same as Series(1) with \<episodeguideurl> removed_. I then loaded Kodi and let it scan.

**Much to my surprise, Series(2) did NOT scan into the library.** o.0 Kodi's log file was unenlightening with the same old "invalid literal for int() with base 10" error.

I went ahead and changed Series(3) to be the same as Series(1) was and gave it a go. 

**That did not fix it.** They did NOT scan in. :(

The only difference I can identify between Series(1) and the other two is that Series(1) did not have any UNWATCHED episodes (and thus was not displayed on Kodi's TV series menu [but does now]) whilst the other two _do_ have unwatched episodes and _do_ appear on Kodi's series menu but will NOT bring these new episodes in.

**Alas, I'm officially stuck.** I cannot think of anything further to try.

**Any suggestions for getting these episodes in?**

**Comment by KarellenX** [comment]
If you would only provide a full log and make it easier for us!!!!

So I ask you clearly **Please provide a full debug log** We do NOT want snippets that you think are important or relevant. 

Also read about the episodeguide url here... 
https://kodi.wiki/view/NFO_files/TV_shows#episodeguide_URL

Also there is no such tag as `<episodeguideurl>`
But there is an `<episodeguide>` tag

**Comment by romanvm** [comment]
I agree, a full debug log is needed.

**Comment by nSpecter** [comment]
Thanks a million for the help. One Kodi debug log comin' up!

[kodi.debug.log.txt](https://github.com/romanvm/kodi.tvmaze/files/4081782/kodi.debug.log.txt)

**Comment by KarellenX** [comment]
Your log paints a different picture to your first post. You are using v19 **and** you are using NFO files both important points which you failed to mention in your first post.

And I still feel this is only part of the picture, because you won't answer questions or comments that have been directed to you.

>I guess you originally scraped your show with TVDB, now you have decided to change scrapers mid-show and are trying to add new episodes to the existing show in your library.

Response required.

>Once you scrape a show with a specific scraper, you need to continue to use it for subsequent scrapes of new episodes of the show. If you want to change scrapers, then you need to Refresh the TV Show (no need to refesh the episodes)

Response required

>Also read about the episodeguide url here...
https://kodi.wiki/view/NFO_files/TV_shows#episodeguide_URL

Did you read this? Did you understand it? Do you have any comments or questions?

>Also there is no such tag as `<episodeguideurl>`
But there is an `<episodeguide>` tag

Typ-o or the cause of your problems?

Your problem should have been raised on the forum. But I will let romanvm decide how to proceed.

**Comment by KarellenX** [comment]
Just a further comment...

You are using Sonarr to create NFO Files, who uses TVDB to scrape metadata from TVDB (Not TVMaze) and another thing that should have been mentioned in the first post.

Sonarr are known for their non-compatible NFO Files and they are still using TVDB v1 API while Kodi uses v2 API and no longer recognises v1.

Read here... https://forum.kodi.tv/showthread.php?tid=323588

And I would appreciate you reading before posting a barrage of further questions/comments.

As far as I can see this is user error and no issue with the scraper.

**Comment by nSpecter** [comment]
My apologies. I'm greatly ignorant to most all of this, so I have little grounds for understanding and parsing what's-what or what is important and what isn't. Please bear with me.

> You are using v19 and you are using NFO files both important points which you failed to mention in your first post.

That is correct. It's not that I won't answer questions, it's just that I failed to understand the significance of it, so it didn't cross my mind to mention it.

> I guess you originally scraped your show with TVDB, now you have decided to change scrapers mid-show and are trying to add new episodes to the existing show in your library.

That is correct. For years I've used TheTVDB but due to periodic issues with their recalcitrant admins, compounded by their first attempt to upgrade their site and its subsequent roll-back, caused me to seek another solution, whereupon I ran across TVMaze in the (Reddit?) forums.

In Kodi I changed the scraper for TV shows to TVMaze and it prompted me whether I wished to rescan the entire library and I responded to the affirmative and left it to do its thing. When I came back I found everything working fine. And it worked fine for a number of months. Up until TheTVDB did their most recent change and started causing spurious errors, both in Sonarr and in Kodi with TVMaze. Which ultimately resulted in this post to you, seeking information/solutions.

> Once you scrape a show with a specific scraper, you need to continue to use it for subsequent scrapes of new episodes of the show. If you want to change scrapers, then you need to Refresh the TV Show (no need to refesh the episodes)

If you are referring to the optional scan performed when you select a new scraper in Kodi and it asks if you want to rescan the library, I did that, it did that and life was peachy for a period of time. If you're _not_ referring to that, then I do not have the foggiest of clues as to what you're referring to when you say, "Refresh the TV Show (no need to refesh the episodes)". ¯\\\_(ツ)_/¯

> [_reference to a wiki page_] Did you read this? Did you understand it? Do you have any comments or questions?

Until now, I did not know of its existence and I dare say I have not read the entirety of the wiki site, cover-to-cover (and likely wouldn't understand most of it). I will give that page a good, solid read forthwith.

> Also there is no such tag as \<episodeguideurl>
>  But there is an \<episodeguide> tag

> Typ-o or the cause of your problems?

I don't know how to respond to that. Other than to say that **ALL** of the tvshow.nfo files I have looked at contain \<episodeguide> AND \<episodeguideurl> as a pair. One after the other, both referencing the same URL. What that means, I have no idea. Where it comes from, I have no idea. Is it normal or an anomaly, I have no idea. This is all Greek to me.

> You are using Sonarr to create NFO Files, who uses TVDB to scrape metadata from TVDB (Not TVMaze) and another thing that should have been mentioned in the first post.

This is all Greek to me.

> Read here...

Will do.

> And I would appreciate you reading before posting a barrage of further questions/comments.

I shall endeavour to do so. And rest assured your annoyance is coming through loud and clear. Also rest assured I _have_ attempted to perform my due diligence with testing, pouring over log files and thoughtful analysis before bringing my problem here. The information at my disposal, primarily the error messages in the log files, combined with a limited understanding of how **any** of this stuff works, lead me inexorably to seek a solution here.

Keep in mind: **YOU** live and breath this stuff. Its every nuance carries meaning and import. Intrepid explorers like me who come here seeking help are likely to be wholly ignorant and drowning in unfamiliar, mysterious detail. A little less brash annoyance would be... friendlier. ;)

Thanks again for your help. It truly is appreciated. Now, I'm off to study those links you gave me and hopefully a solution to this mysterious quandary is just over the horizon.

TTFN :)

**Comment by KarellenX** [comment]
No need to read the wiki from cover to cover, I only point you to pages that are relevant to the issue at hand, and only so I don't have to repeat myself. Sure I can copy and paste sections of the wiki into a post perfectly curated for you, but you could just as easily click the link and read the section I pointed to.

>Keep in mind: YOU live and breath this stuff. 

Well forgive me because the way you wrote and worded your first post with your extracts of errors, I had the impression you were knowledgeable. So if you had no idea what you were doing, then provide the full log. Was I annoyed... yes. Because I asked once for a debuglog and you did not provide it. I asked you some questions, and there was not one word back from you in feedback that indicated whether what I referred to was or was not part of the problem, so you kept responding but being completely unhelpful.

>have looked at contain `<episodeguide>` AND `<episodeguideurl>` as a pair.

That is not Kodi. I guess this is Sonarr creating non-compliant tags, most likely to accommodate another media centre that uses that tag. Must admit I have not seen its use before. Kodi uses `<episodeguide>`

>in Kodi I changed the scraper for TV shows to TVMaze and it prompted me whether I wished to rescan the entire library and I responded to the affirmative and left it to do its thing. When I came back I found everything working fine. 

I think something has gone wrong there because it is clear that the Kodi database still holds TVDB references instead of TVMaze references. But your use of Sonar NFO files is most likely responsible for this. If you have nfo files in your folder, then no matter what scraper you use, and no matter how many times you try to scrape, the NFO Files will be scanned because Kodi uses local files **first. Always!** So you may have rescanned and let Kodi "do its thing" and its thing was to rescan NFO files, so nothing changed.

So I must admit I am confused about your workflow. You obviously use Sonarr to create nfo files, so what is the deal with trying to change scrapers? If you have nfo files, no scrapers will be used apart from some initial checks in the scanning process. Certainly nothing will be scraped from the online sources.

**Comment by nSpecter** [comment]
> So I must admit I am confused about your workflow. 

You and me, both. :)

> You obviously use Sonarr to create nfo files, so what is the deal with trying to change scrapers? If you have nfo files, no scrapers will be used apart from some initial checks in the scanning process. Certainly nothing will be scraped from the online sources.

All I know is, though I've been a Kodi user for years, this particular Kodi install is largely a default installation per a typical walk-through posted on the Internet, It's even still using the Confluence skin.

Sonarr is largely a default installation (per walk-throughs) as is Radarr and they just do their things and Sonarr moves downloaded episodes to the appropriate volume/directory with an NFO file. And there it sits forever--_with Kodi none the wiser_-- **UNTIL** you either,

- Go into TV Shows -> Options -> "Update Library" _or_
- Turn on System -> Media -> Library -> Video Library -> "Update library on Startup" (which is what I have)

If you don't "Update Library", Kodi never sees the new episodes in the Series/Season folder. And that's where the TVMaze (or TVDB) scraper comes in. "Update Library" scans for new content and imports it into Kodi's database, yes? And it scrapes JPGs and banners and fanart and what-have-you, yes?

That's what is not happening.  _For just these few shows._ 

For other shows, new episodes appear to be importing just fine with TVMaze. I just now looked at a few recently imported episodes and those shows have the exact same tvshow.nfo XML lines as the shows that TVMaze chokes on. But the scraper (_or whatever this problem is_) doesn't appear to be having a problem with those shows. So, why these? ¯\\\_(ツ)_/¯

**Comment by KarellenX** [comment]
>"Update Library" scans for new content and imports it into Kodi's database, yes? And it scrapes JPGs and banners and fanart and what-have-you, yes?

Yes and yes.

>For other shows, new episodes appear to be importing just fine with TVMaze. I just now looked at a few recently imported episodes and those shows have the exact same tvshow.nfo XML lines as the shows that TVMaze chokes on. But the scraper (or whatever this problem is) doesn't appear to be having a problem with those shows. So, why these? ¯\_(ツ)_/¯

Ok, the crux of the problem. So while the debug log helps narrowing down the problem, I will need to see the following...
1. tvshow.nfo file from a tvshow that allows scanning of new episodes 
2. tvshow.nfo file from a tvshow that "chokes" when adding new episodes.
Both pasted to https://paste.kodi.tv/ separately
3. Your MyVideosxx.db database, zipped and uploaded to a file sharing service so I can download.

Let's not forget that Sonarr/Radar have non-compatible files and are still using the v1 API. But I would like to see why some shows work, and some don't.

**Comment by nSpecter** [comment]
>1. tvshow.nfo file from a tvshow that allows scanning of new episodes
These two imported fine two days ago,
[Law & Order:SVU](https://paste.kodi.tv/nusuxubuxu)
[QI](https://paste.kodi.tv/pakaterero)

>2.  tvshow.nfo file from a tvshow that "chokes" when adding new episodes.
These two produce errors and have numerous unimported episodes pending
[Grantchester](https://paste.kodi.tv/ogapiretaj)
[Vikings](https://paste.kodi.tv/rewesakuqu)

[MyVideos116.db.zip](https://mega.nz/#!WJB0yaQC!Mj0lYiWhaM-3cT-TyxKXFw-XPHqcvVb3R0q61rI9Krw)

**Comment by KarellenX** [comment]
Thanks. I see Sonarr have fixed their nfo files since the last time I looked at them... except for the episodguide which still pretty much makes them useless.

So you have no TVMaze episode guides for any show in the database. 

You have a few tv shows with the correct v2 API for the TVDB (green). These will work if you switch back to the TVDB scraper. All you other shows are v1 and won't work (red). See image below. None of your shows will work with TV Maze scraper as they are now.

![episodeguide](https://user-images.githubusercontent.com/44626605/72674351-768cd500-3ac9-11ea-8172-fc2cae5c3d01.jpg)

So like I said in my very first post to you, you cannot switch scrapers mid-scrape. You cannot use TV Maze with TVDB links. They are two completely different sites completely unrelated to each other.

As for why one show works and another doesn't, no idea. I cannot figure this out remotely and with the peculiar setup you have. All I can tell you is to stop using Sonarr nfo files, and just scrape your tv shows using Kodi. Most of the episodeguides are wrong and need to be replaced and how to do that is explained in one of the links I gave you, and in one of my previous posts.
